packages/host/app/components/matrix/room-message.gts

```diff
@@ -231,24 +243,37 @@ export default class RoomMessage extends Component<Signature> {
     await navigator.clipboard.writeText(this.previewPatchCode);
   });
 
-  private get resources() {
+  private loadMessageResources = trackedFunction(this, async () => {
     let cards: CardDef[] = [];
     let errors: { id: string; error: Error }[] = [];
-    this.args.message.attachedResources?.map((resource) => {
-      if (resource.card) {
-        cards.push(resource.card);
-      } else if (resource.cardError) {
-        let { id, error } = resource.cardError;
-        errors.push({
-          id,
-          error,
-        });
-      }
-    });
+
+    let promises = this.args.message.attachedResources?.map(
+      async (resource) => {
+        await resource.loaded;
+        if (resource.card) {
+          cards.push(resource.card);
+        } else if (resource.cardError) {
+          let { id, error } = resource.cardError;
+          errors.push({
+            id,
+            error,
+          });
+        }
+      },
+    );
+
+    if (promises) {
+      await Promise.all(promises);
+    }
+
     return {
       cards: cards.length ? cards : undefined,
       errors: errors.length ? errors : undefined,
     };
+  });
+
+  private get resources() {
+    return this.loadMessageResources.value;
```

Comment:
Is `.value` awaited automatically?? I see in the types that there is other api (eg `.isFinished`) that reveal the resolution of the promise but if `.value` awaits it then we don't need to worry.

Reply:
That's right, at first `this.loadMessageResources.value` is `undefined` and will get a value after `loadMessageResources` is resolved.